### ISSUE TITLE
mb/mh write incorrect values

### ISSUE THREAD

**Issue by bereziat** [heimdali/heimdali#38]
Currently, mh and mb commands write a file with -f -o 1 coding value. That's mean the output values
should be 0 or 255 but they are 0 or 1 that is incorrect.

`$ mh -n 0.5 lena.h5 | par`
`ghost.inr x 256    -y 256  -o 1    -f`
`$ mh -n 0.5 lena.h5 ls.h5`
`$ tpr -c ls.h5 | head`
`0 0 0 0 0`
`0 0 0 0 0`
`0 0 0 0 0`
`0 0 0 0 0`
`0 0 0 0 0`
`0 0 0 0 0`
`0 1 1 1 1`
`1 1 1 1 1`
`1 1 1 1 1`
`1 1 1 1 1`

But the problem is perhaps elsewhere:
`$ mh -n 0.5 lena.h5 | tpr`
`plane 0, line 0`
`0`
`plane 0, line 1`
`0`
`....`
`plane 0, line 46`
`0`
`plane 0, line 47`
`255`
`plane 0, line 48`
`255`
`...`
`plane 0, line 255`
`255`
`plane 1, line 0`
`Segmentation fault: 11`

In this last example, coordinate are buggy, but values are correct. And tpr crashes...

**Comment by dfroger** [comment]
tpr crash is fixed in 3081b6bbe49125204b9802784a53dbe43aa53ffb

**Comment by dfroger** [comment]
`mh` and `mb` actually write 0 or 255.

Problem is with `tpr`. 

How the original `tpr` decide to print using the fixed-point or floating-point representation seems a bit inconsitent to me:

```
tpr -x 1 -y 1 -c lena_f1.inr
165

tpr -f="%d" -x 1 -y 1 -c lena_f1.inr
165

tpr -f="%g" -x 1 -y 1 -c lena_f1.inr
0.647059

tpr -x 1 -y 1 -c lena_f2.inr
0.647059

tpr -f="%d" -x 1 -y 1 -c lena_f2.inr
42405

tpr -f="%g" -x 1 -y 1 -c lena_f2.inr
0.647059
```

**Comment by dfroger** [comment]
The now clear with the [doc](http://inrimage.gforge.inria.fr/WWW/tpr.1i.html)

**Comment by dfroger** [comment]
Fixed in 48f086e
